src/app/query/stacks/balance/stacks-ft-balances.hooks.ts

```diff
@@ -90,3 +93,41 @@ export function useTransferableStacksFungibleTokenAssetBalances(
     [assetBalances]
   );
 }
+
+/**
+ * @see https://github.com/leather-wallet/issues/issues/16
+ */
+type StacksFtTokensFilter = 'all' | 'supported' | 'unsupported';
+
+interface useFilteredStacksFungibleTokenListArgs {
+  address: string;
+  filter?: StacksFtTokensFilter;
+}
+export function useFilteredStacksFungibleTokenList({
+  address,
+  filter = 'all',
+}: useFilteredStacksFungibleTokenListArgs) {
+  const stacksFtAssetBalances = useStacksFungibleTokenAssetBalancesWithMetadata(address);
+  const { data: swapAssets = [] } = useAlexSwappableAssets();
+
+  return useMemo(() => {
+    if (filter === 'supported') {
+      return stacksFtAssetBalances.filter(assetBalance =>
+        swapAssets.some(swapAsset =>
+          swapAsset.principal.includes(assetBalance.asset.contractAddress)
+        )
+      );
+    }
+
+    if (filter === 'unsupported') {
+      return stacksFtAssetBalances.filter(
+        assetBalance =>
+          !swapAssets.some(swapAsset =>
+            swapAsset.principal.includes(assetBalance.asset.contractAddress)
+          )
+      );
+    }
+
+    return stacksFtAssetBalances;
+  }, [stacksFtAssetBalances, swapAssets, filter]);
+}
```

Comment:
Can we break apart the filtering from the hook that binds it to the state?

Reply:
separated it